securedrop/journalist_app/api2/__init__.py

```diff
@@ -131,11 +93,27 @@ def metadata() -> Response:
     the ``Reply`` tables for the set of all item UUIDs.
     """
     try:
-        requested = MetadataRequest(**request.json)  # type: ignore
+        requested = BatchRequest(**request.json)  # type: ignore
     except (TypeError, ValueError) as exc:
         abort(422, f"malformed request; {exc}")
 
-    response = MetadataResponse()
+    response = BatchResponse()
+
+    for event in requested.events:
+        """
+        # Case 1: already processed: If event.snowflake_id is already cached in
+        # Redis, we've already processed it; just return its status.
+        status = EventStatus(event.snowflake_id, 200)  # FIXME
+        response.events.append(status)
+        """
+
+        # Case 2: needs to be processed.
+        result = EventHandler.process(event)
+        for uuid, source in result.sources.items():
+            response.sources[uuid] = source.to_api_v2()
+        for uuid, item in result.items.items():
+            response.items[uuid] = item.to_api_v2()
+        response.events[result.event_id] = result.status
```

Comment:
This is the entry-point for handling events.  The `events` dictionary is just status information.  Changes are returned in the `sources` and `items` dictionaries just like reads.